nft_ingester/src/bin/migrator/main.rs

```diff
@@ -206,61 +202,62 @@ impl JsonMigrator {
 
-                let end_point = {
-                    if tasks_buffer.len() < tasks_batch_to_insert {
-                        tasks_buffer.len()
-                    } else {
-                        tasks_batch_to_insert
+                    if tasks_buffer.is_empty() {
+                        drop(tasks_buffer);
+                        tokio::time::sleep(tokio::time::Duration::from_secs(10)).await;
+                        continue;
                     }
-                };
 
-                let mut tasks_to_insert = tasks_buffer.drain(0..end_point).collect::<Vec<Task>>();
+                    let end_point = {
+                        if tasks_buffer.len() < tasks_batch_to_insert {
+                            tasks_buffer.len()
+                        } else {
+                            tasks_batch_to_insert
+                        }
+                    };
 
-                drop(tasks_buffer);
+                    let mut tasks_to_insert =
+                        tasks_buffer.drain(0..end_point).collect::<Vec<Task>>();
 
-                let res = cloned_pg_client.insert_json_download_tasks(&mut tasks_to_insert).await;
-                match res {
-                    Ok(_) => {
-                        cloned_metrics.inc_tasks_set(
-                            "tasks_set",
-                            MetricStatus::SUCCESS,
-                            tasks_to_insert.len() as u64,
-                        );
-                    },
-                    Err(e) => {
-                        cloned_metrics.inc_tasks_set(
-                            "tasks_set",
-                            MetricStatus::FAILURE,
-                            tasks_to_insert.len() as u64,
-                        );
-                        error!("insert_tasks: {}", e)
-                    },
+                    drop(tasks_buffer);
+
+                    let res =
+                        cloned_pg_client.insert_json_download_tasks(&mut tasks_to_insert).await;
+                    match res {
+                        Ok(_) => {
+                            cloned_metrics.inc_tasks_set(
+                                "tasks_set",
+                                MetricStatus::SUCCESS,
+                                tasks_to_insert.len() as u64,
+                            );
+                        },
+                        Err(e) => {
+                            cloned_metrics.inc_tasks_set(
+                                "tasks_set",
+                                MetricStatus::FAILURE,
+                                tasks_to_insert.len() as u64,
+                            );
+                            error!("insert_tasks: {}", e)
+                        },
+                    }
                 }
             }
-
-            Ok(())
         });
 
         assets_iter.seek_to_first();
         while assets_iter.valid() {
-            if !rx.is_empty() {
+            if cancellation_token.is_cancelled() {
```

Comment:
The body will block the cancellation Maybe it's intended

Reply:
will be reviewed in the next iterations of this PR